test/e2e/components/vl-rich-data.js

```diff
@@ -0,0 +1,53 @@
+const {VlElement} = require('vl-ui-core').Test;
+const {By} = require('vl-ui-core').Test.Setup;
+const {VlPager} = require('vl-ui-pager').Test;
+const {VlSearchFilter} = require('vl-ui-search-filter').Test;
+const {VlModal} = require('vl-ui-modal').Test;
+
+class VlRichData extends VlElement {
+  async getContentSlotElements() {
+    return this._getSlotElements('content');
+  }
+
+  async getSearchFilter() {
+    const searchFilter = await this.findElement(By.css('[is="vl-search-filter"]'));
+    return new VlSearchFilter(this.driver, searchFilter);
+  }
+
+  async getPager() {
+    const assignedElements = await this._getPagerSlotElements();
+    return new VlPager(this.driver, assignedElements[0]);
+  }
+
+  async toggleSearchFilter() {
+    const button = await this.shadowRoot.findElement(By.css('#toggle-filter-button'));
+    return button.click();
+  }
+
+  async closeSearchFilter() {
+    const button = await this.shadowRoot.findElement(By.css('#close-filter-button'));
+    return button.click();
+  }
+
+  async openModalSearchFilter() {
+    const button = await this.shadowRoot.findElement(By.css('#open-filter-button'));
+    return button.click();
+  }
+
+  async closeModalSearchFilter() {
+    const modalElement = await this.shadowRoot.findElement(By.css('#filter-modal'));
+    const modal = await new VlModal(this.driver, modalElement);
```

Comment:
deze await mag weg voor de new. Als je new gebruikt als keyword, verwacht je altijd direct een object terug. 

Ik vermoed dat je met es5 prototype constructs wel een async constructor function kunt maken. Maar lijkt mij niet direct de bedoeling.

Reply:
Voor de e2e testen maken we achterliggend gebruik van de Selenium Webdriver elementen om elegant met selectors constructies te realiseren. Achterliggend wordt de `driver.findElement` gebruikt waardoor de constructie van e2e page objecten async is.